WMF Framework/UIScrollView+Limits.swift

```diff
@@ -0,0 +1,46 @@
+import UIKit
+
+extension UIScrollView {
+    fileprivate var wmf_topOffsetY: CGFloat {
+        return 0 - contentInset.top
+    }
+
+    fileprivate var wmf_bottomOffsetY: CGFloat {
+        return contentSize.height - bounds.size.height + contentInset.bottom
+    }
+
+    fileprivate var wmf_topOffset: CGPoint {
+        return CGPoint(x: contentOffset.x, y: wmf_topOffsetY)
+    }
+
+    fileprivate var wmf_bottomOffset: CGPoint {
+        return CGPoint(x: contentOffset.x, y: wmf_bottomOffsetY)
+    }
+
+    fileprivate var wmf_isAtTop: Bool {
+        return contentOffset.y <= wmf_topOffsetY
+    }
+
+    fileprivate var wmf_isAtBottom: Bool {
+        return contentOffset.y >= wmf_bottomOffsetY
+    }
+
+    @objc public func wmf_setContentInsetPreservingTopAndBottomOffset(_ updatedContentInset: UIEdgeInsets, scrollIndicatorInsets updatedScrollIndicatorInsets: UIEdgeInsets, withNavigationBar navigationBar: NavigationBar?) -> Bool {
+        guard updatedContentInset != contentInset || updatedScrollIndicatorInsets != scrollIndicatorInsets else {
+            return false
+        }
+        let wasAtTop = wmf_isAtTop
+        let wasAtBottom = wmf_isAtBottom
+        scrollIndicatorInsets = updatedScrollIndicatorInsets
+        contentInset = updatedContentInset
+        if wasAtTop {
+            contentOffset = wmf_topOffset
+        } else if wasAtBottom {
+            contentOffset = wmf_bottomOffset
+        }
+        if wmf_isAtTop {
```

Comment:
Is *is* (vs *was*) intentional here?

Reply:
Yes, sometimes the rotation will put it at the top even if it wasn't at the top. On an X in portrait, view explore feed, scroll just the search bar invisible, rotate to landscape.